applications/jupyter/README.md

```diff
@@ -2,125 +2,91 @@
 
-9. After installing Jupyterhub, you will need to retrieve the name of the backend-service from GCP using the following command:
+4. Run `terraform init`
 
-    ```cmd
-    gcloud compute backend-services list --project=%PROJECT_ID%
-    ```
+5. Run `terraform apply --var-file=./workloads.tfvars`
 
-    You can describe the service to check which service is connected to `proxy-public` using:
+> **_NOTE:_** To secure the Jupyter endpoint, this module enables IAP by default. It is _strongly recommended_ to keep this configuration. If you wish to disable it, do the following: set the `add_auth` flag to false in the `variable.tf` file.
 
-    ```cmd
-    gcloud compute backend-services describe SERVICE_NAME --project=%PROJECT_ID% --global
-    ```
+> **_NOTE:_** Wait for terraform apply to complete successfully & then rerun it again. This is needed to configure Jupyter with IAP correctly. 
 
-10. Once you get the name of the backend-service, replace the variable in the [variables.tf](https://github.com/GoogleCloudPlatform/ai-on-gke/blob/main/jupyter-on-gke/variables.tf) file.
+6. Run `terraform apply --var-file=./workloads.tfvars`
 
-11. Re-run `terraform apply`
+7. Note down the value for `domain` from terraform output section. You can open this in a browser & login with your credentials 
 
-12. Navigate to the [GCP IAP Cloud Console](https://console.cloud.google.com/security/iap) and select your backend-service checkbox.
+> **_NOTE:_** Domain specefic managed certificate may take some time to finish provisioning. On average around 10-15 minutes. 
 
-13. Click on `Add Principal`, insert the new principle and select under `Cloud IAP` with role `IAP-secured Web App User`
+8. Navigate to the [GCP IAP Cloud Console](https://console.cloud.google.com/security/iap) and select your backend-service checkbox.
 
-> **_NOTE:_** Your managed certificate may take some time to finish provisioning. On average around 10-15 minutes. 
+9. Click on `Add Principal`, insert the new principle and select under `Cloud IAP` with role `IAP-secured Web App User`
 
 ## Using Jupyterhub
 
 ### If Auth is disabled
 
-1. Run `kubectl get services -n <namespace>`. The namespace is the user name that you specified above.
-
-2. Copy the external IP for the notebook. If there isn't an external IP for `proxy-public`, is it most likely due to authencation being enabled.
+1. Visit [Services](https://console.cloud.google.com/kubernetes/discovery) section on the GKE console & open the external IP for the `proxy-public` service in the browser.
 
-Continue to Step 3 of [below](#if-auth-is-enabled).
+> **_NOTE:_** If there isn't an external IP for `proxy-public`, is it most likely due to authencation being enabled.
 
 ### If Auth is enabled
 
-1. Run `kubectl describe managedcertificate managed-cert -n <namespace>`. The namespace is the user name that you specified above.
-
-2. Copy the domain under `Domains`
+1. Copy the value for `domain` from terraform output section.
```

Comment:
what is terraform output section? is that what is printed after running terraform apply based on outputs.tf? What if they come back to this at a later date, how do they find the domain? Is it visible anywhere in pantheon?

Reply:
Added reference to previous step & also to the managed certificate page where domain value can be obtained from